File: AndroidPatternsApp/jmvp-app/src/main/java/com/woowrale/jmvp/di/components/MainComponent.java
package com.woowrale.jmvp.di.components;

import com.woowrale.jmvp.ui.base.BaseActivity;
import com.woowrale.jmvp.ui.search.home.SearchActivity;
import com.woowrale.jmvp.ui.search.local.LocalSearchActivity;
import com.woowrale.jmvp.ui.search.remote.RemoteSearchActivity;
import com.woowrale.jmvp.di.modules.MainModule;
import com.woowrale.jmvp.di.modules.ServicesModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {MainModule.class, ServicesModule.class})
public interface MainComponent {

    void inject(BaseActivity activity);

    void inject(SearchActivity activity);

    void inject(LocalSearchActivity activity);

    void inject(RemoteSearchActivity activity);

}

File: AndroidXPatternsApp/kmvvm-app/src/main/java/com/woowrale/kmvvm/di/components/MainComponent.kt
package com.woowrale.kmvvm.di.components

import com.woowrale.kmvvm.di.modules.MainModule
import com.woowrale.kmvvm.di.modules.ServicesModule
import com.woowrale.kmvvm.ui.base.BaseActivity
import com.woowrale.kmvvm.ui.search.home.SearchActivity
import com.woowrale.kmvvm.ui.search.local.LocalSearchActivity
import com.woowrale.kmvvm.ui.search.remote.RemoteSearchActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [MainModule::class, ServicesModule::class])
interface MainComponent {

    fun inject(activity: BaseActivity)

    fun inject(activity: SearchActivity)

    fun inject(activity: LocalSearchActivity)

    fun inject(activity: RemoteSearchActivity)

}